python/graphscope/nx/algorithms/builtin.py

```diff
@@ -1184,12 +1184,158 @@ def is_simple_path(G, nodes):
     False
 
     """
-    if isinstance(nodes, list):
-        n1json = json.dumps(nodes)
-        ctx = AppAssets(algo="is_simple_path", context="tensor")(G, n1json)
-        return ctx.to_numpy("r", axis=0)[0]
+    return graphscope.is_simple_path(G, nodes)
+
+
+@project_to_simple
+def all_simple_paths(G, source, target_nodes, cutoff=None):
+    """Generate all simple paths in the graph G from source to target.
+    A simple path is a path with no repeated nodes.
+    Parameters
+    ----------
+    G : NetworkX graph
+    source : node
+       Starting node for path
+    target : nodes
+       Single node or iterable of nodes at which to end path
+    cutoff : integer, optional
+        Depth to stop the search. Only paths of length <= cutoff are returned.
+    Returns
+    -------
+    path_generator: generator
+       A generator that produces lists of simple paths.  If there are no paths
+       between the source and target within the given cutoff the generator
+       produces no output.
+    Examples
+    --------
+        >>> G = nx.complete_graph(4)
+        >>> print(nx.builtin.all_simple_paths(G, 0, 3))
+        ...
+        [0, 1, 2, 3]
+        [0, 1, 3]
+        [0, 2, 1, 3]
+        [0, 2, 3]
+        [0, 3]
+    Notes
+    -----
+    This algorithm uses a modified depth-first search to generate the
```

Comment:
We not implement all_simple_paths like networkx, so the Notes can remove or modify ?